client/landing/stepper/declarative-flow/site-migration-flow.ts

```diff
@@ -41,7 +42,7 @@ const siteMigration: Flow = {
 			? [ STEPS.PICK_SITE, STEPS.SITE_CREATION_STEP, STEPS.PROCESSING ]
 			: [];
 
-		return [ ...baseSteps, ...hostedVariantSteps ];
+		return stepsWithRequiredLogin( [ ...baseSteps, ...hostedVariantSteps ] );
 	},
 
 	useAssertConditions(): AssertConditionResult {
```

Comment:
We should remove the logged in checks and redirects from `useAssertConditions()` so we're more confident that our new handling is doing what it should.